src/Makevars

```diff
@@ -1,2 +1,2 @@
-CXX_STD = -std=c++17
-CXX17 = -std=c++17
\ No newline at end of file
+CXX_STD = -std=gnu++17
+CXX17 = -std=gnu++17
```

Comment:
@diego-urgell this seems incorrect. According to https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Using-C_002b_002b-code  you should instead use 
```
CXX_STD = CXX17
```
and then R will generate the compiler-specific flags (such as `-std=gnu++17`) that are required to use the C++ 17 standard.

Reply:
Yes, you are right. Originally it was like that but I changed to test if I could "hardcode" the standard. However, it still generates the same error. I will open a new branch to remove all C++14 dependencies and see if the package compiles successfully with g++ that way.

Reply:
Also, I thought that maybe it was because of the R version, but even after specifying the latest release, the error persists. I also got the same result using an Azure VM, so the problem is likely because of the new C++ features :(

Reply:
before you remove the C++14 / 17 features, please post to R-package-devel with sufficient details (minimal reproducible example) to explain your problem. R says it supports C++14 / 17, so you should be able to get this to work. (but there may be some missing configuration, or maybe a bug in R)
If you don't get a good response from R-pacakge-devel, it may be appropriate for https://stat.ethz.ch/mailman/listinfo/r-devel

Reply:
also you should try to compile a source code "by hand" using gcc. can you install gcc on your computer? can you try to manually write a command for compiling one of your C++17 files with gcc? If you can get gcc to do it, you should be able to tell R how to generate that command.